pkg/operation/botanist/component/logging/event_logger.go

```diff
@@ -0,0 +1,364 @@
+type Interface interface {
+	component.Deployer
+}
+
+type eventLogger struct {
+	client           client.Client
+	namespace        string
+	imageEventLogger string
+	secretsManager   secretsmanager.Interface
+}
+
+// NewEventLogger creates a new instance of event-logger for the shoot event logging.
+func NewEventLogger(
+	client client.Client,
+	namespace string,
+	secretsManager manager.Interface,
+	imageEventLogger string,
+) (Interface, error) {
+	if client == nil {
+		return nil, errors.New("client cannot be nil")
+	}
+
+	if len(namespace) == 0 {
+		return nil, errors.New("namespace cannot be empty")
+	}
+
+	if len(imageEventLogger) == 0 {
+		return nil, errors.New("imageEventLogger cannot be empty")
+	}
+
+	if secretsManager == nil {
+		return nil, errors.New("secret manager cannot be nil")
+	}
+
+	return &eventLogger{
+		client:           client,
+		namespace:        namespace,
+		imageEventLogger: imageEventLogger,
+		secretsManager:   secretsManager,
+	}, nil
+}
+
+func (l *eventLogger) Deploy(ctx context.Context) error {
+	var (
+		serviceAccount = l.emptyServiceAccount()
+		role           = l.emptyRole()
+		roleBinding    = l.emptyRoleBinding()
+		deployment     = l.emptyDeployment()
+	)
+
+	if err := l.deployEventLoggerRBACForShootAPI(ctx); err != nil {
+		return err
+	}
+
+	// Deploy Event Logger RBAC for the Seed API
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, serviceAccount, func() error {
+		serviceAccount.AutomountServiceAccountToken = pointer.BoolPtr(false)
+		serviceAccount.Labels = getEventLoggerLabels()
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, role, func() error {
+		role.Labels = getEventLoggerLabels()
+		role.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"events"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, roleBinding, func() error {
+		roleBinding.Labels = getEventLoggerLabels()
+		roleBinding.RoleRef = rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     role.Name,
+		}
+		roleBinding.Subjects = []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      serviceAccount.Name,
+			Namespace: l.namespace,
+		}}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	genericTokenKubeconfigSecret, found := l.secretsManager.Get(v1beta1constants.SecretNameGenericTokenKubeconfig)
+	if !found {
+		return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameGenericTokenKubeconfig)
+	}
+
+	// Deploy the Event Logger into the shoot control-plane
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, deployment, func() error {
+		deployment.Labels = getEventLoggerLabels()
+		deployment.Spec.RevisionHistoryLimit = pointer.Int32Ptr(1)
```

Comment:
```suggestion
		deployment.Spec.RevisionHistoryLimit = pointer.Int32(1)
```
`pointer.Int32Ptr` is deprecated.